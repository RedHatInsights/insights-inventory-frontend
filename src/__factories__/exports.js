import { faker } from '@faker-js/faker';
import {
  EXPORT_SERVICE_APPLICATON,
  EXPORT_SERVICE_RESOURCE,
} from '../components/InventoryTable/hooks/useInventoryExport/constants';

const exportObject = ({ format, status } = {}) => ({
  id: faker.string.uuid(),
  name: faker.lorem.words({ min: 1, max: 1 }),
  created_at: faker.date.past({ years: 2 }).toISOString(),
  completed_at: faker.date.past({ years: 1 }).toISOString(),
  expires_at: faker.date.future({ years: 1 }).toISOString(),
  format,
  status,
  sources: [
    {
      application: EXPORT_SERVICE_APPLICATON,
      resource: EXPORT_SERVICE_RESOURCE,
      filters: {},
      id: faker.string.uuid(),
      status,
    },
  ],
});

export default (count = 1, properties = {}) =>
  [...Array(count)].map(() => exportObject(properties));
