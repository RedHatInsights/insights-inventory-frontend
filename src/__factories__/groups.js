import { faker } from '@faker-js/faker';

export const buildGroups = (length, allEmpty, allNonEmpty) => {
  return [...Array(length)].map(() => {
    const created_at = faker.date.past({ years: 3 }).getTime();

    return {
      updated_at: faker.date.soon({ refDate: created_at }).getTime(),
      id: faker.string.uuid(),
      created_at,
      name: faker.lorem.words({ min: 1, max: 3 }),
      host_count: allEmpty
        ? 0
        : allNonEmpty
        ? faker.number.int({ min: 1, max: 100 })
        : faker.number.int({ min: 0, max: 100 }),
    };
  });
};

export const buildGroupsPayload = (
  page = 1,
  per_page = 50,
  total = 20,
  allEmpty = false,
  allNonEmpty = false
) => {
  const count = Math.min(page * per_page, total);

  return {
    count,
    page,
    per_page,
    total,
    results: buildGroups(count, allEmpty, allNonEmpty),
  };
};
