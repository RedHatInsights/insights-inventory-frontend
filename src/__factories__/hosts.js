import { faker } from '@faker-js/faker';
import { sample } from 'lodash';

const OS_VERSIONS = ['CentOS Linux', 'RHEL'];

export const buildHosts = (length) =>
  [...Array(length)].map(() => ({
    id: faker.string.uuid(),
    insights_id: faker.lorem.slug({ min: 1, max: 3 }),
    display_name: faker.lorem.word(),
    updated: faker.date.past({ years: 3 }),
    system_profile: {
      operating_system: {
        name: sample(OS_VERSIONS),
        major: faker.number.int({ min: 1, max: 10 }),
        minor: faker.number.int(50),
      },
    },
    groups:
      Math.random() > 0.5
        ? [
            {
              id: faker.string.uuid(),
              name: faker.lorem.words({ min: 1, max: 3 }),
            },
          ]
        : [],
  }));

export const buildHostsPayload = (page = 1, per_page = 50, total = 20) => {
  const count = Math.min(page * per_page, total);

  return {
    count,
    page,
    per_page,
    total,
    results: buildHosts(count),
  };
};
