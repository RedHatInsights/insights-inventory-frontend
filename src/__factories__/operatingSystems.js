import { faker } from '@faker-js/faker';

export const buildOperatingSystems = (
  count = 10,
  { osName = 'RHEL 7', major = 7 },
) => {
  let minor = -1;
  return [...Array(count)].map(() => {
    minor++;
    return {
      value: {
        name: osName,
        major,
        minor,
      },
      count: faker.number.int({ min: 1, max: 100 }),
    };
  });
};
