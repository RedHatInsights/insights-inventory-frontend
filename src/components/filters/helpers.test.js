import moment from 'moment';
import { buildOperatingSystems } from '../../__factories__/operatingSystems';
import {
  appendGroupSelection,
  fromValidator,
  toOsFilterGroups,
  toValidator,
} from './helpers';

describe('Validates datepicker dates', () => {
  it('ToValidator', () => {
    const startOfToday = moment().startOf('day');
    const endOfToday = moment().endOf('day');
    expect(toValidator(startOfToday)(endOfToday)).toEqual('');
  });
  it('FromValidator', () => {
    const startOfToday = moment().startOf('day');
    const endOfToday = moment().endOf('day');
    expect(fromValidator(endOfToday)(startOfToday)).toEqual('');
  });
});

describe('toOsFilterGroups', () => {
  it('return "No versions available" when OSes are not loaded', () => {
    const operatingSystems = [];

    expect(toOsFilterGroups(operatingSystems, false)).toEqual([
      { items: [{ isDisabled: true, label: 'No versions available' }] },
    ]);
  });

  it('returns ... when OSes are loaded', () => {
    const operatingSystems = [
      ...buildOperatingSystems(20, { osName: 'RHEL', major: 7 }),
      ...buildOperatingSystems(20, { osName: 'RHEL', major: 8 }),
      ...buildOperatingSystems(20, { osName: 'RHEL', major: 9 }),
      ...buildOperatingSystems(20, { osName: 'CentOS Linux', major: 7 }),
      ...buildOperatingSystems(20, { osName: 'CentOS Linux', major: 8 }),
    ];

    expect(toOsFilterGroups(operatingSystems, true)).toMatchSnapshot();
  });

  it('returns os versions ordered descending', () => {
    const operatingSystems = [
      ...buildOperatingSystems(5, { osName: 'RHEL', major: 7 }),
    ];
    const filterGroups = toOsFilterGroups(operatingSystems, true)[0];

    expect(filterGroups.items[0].minor).toEqual(4);
    expect(filterGroups.items[filterGroups.items.length - 1].minor).toEqual(0);
  });
});

describe('appendGroupSelection', () => {
  const operatingSystemsGroups = toOsFilterGroups(
    [
      ...buildOperatingSystems(10, { osName: 'RHEL', major: 7 }),
      ...buildOperatingSystems(10, { osName: 'RHEL', major: 8 }),
      ...buildOperatingSystems(5, { osName: 'RHEL', major: 9 }),
      ...buildOperatingSystems(10, { osName: 'CentOS Linux', major: 7 }),
      ...buildOperatingSystems(10, { osName: 'CentOS Linux', major: 8 }),
      ...buildOperatingSystems(5, { osName: 'CentOS', major: 9 }),
      ...buildOperatingSystems(4, { osName: 'Alma Linux', major: 8 }),
    ],
    true
  );

  it('returns a new selection', () => {
    const selection = {
      'RHEL-8': {
        'RHEL-8-8.8': true,
        'RHEL-8-8.7': true,
      },
      'RHEL-7': {
        'RHEL-7-7.8': true,
        'RHEL-7': null,
        'RHEL-7-7.7': false,
      },
      'CentOS-Linux-7': {
        'CentOS-Linux-7-7.8': true,
        'CentOS-Linux-7': true,
      },
      'Alma-Linux-8': {
        'Alma-Linux-8-8.1': true,
        'Alma-Linux-8': false,
      },
    };

    expect(
      appendGroupSelection(selection, operatingSystemsGroups)['RHEL-8'][
        'RHEL-8'
      ]
    ).toEqual(null);

    expect(
      appendGroupSelection(selection, operatingSystemsGroups)['RHEL-7'][
        'RHEL-7-7.7'
      ]
    ).toEqual(undefined);

    expect(
      Object.keys(
        appendGroupSelection(selection, operatingSystemsGroups)[
          'CentOS-Linux-7'
        ]
      ).length
    ).toEqual(11);

    expect(
      appendGroupSelection(selection, operatingSystemsGroups)['CentOS-Linux-7'][
        'CentOS-Linux-7-7.7'
      ]
    ).toEqual(true);
    console.log(
      appendGroupSelection(selection, operatingSystemsGroups)['Alma-Linux-8']
    );
    expect(
      appendGroupSelection(selection, operatingSystemsGroups)['Alma-Linux-8']
    ).toEqual({ 'Alma-Linux-8': null, 'Alma-Linux-8-8.1': true });
  });

  it('returns a new selection with whole group appended', () => {
    const selection = {
      'RHEL-8': {
        'RHEL-8': true,
      },
    };

    expect(
      Object.keys(
        appendGroupSelection(selection, operatingSystemsGroups)['RHEL-8']
      ).length
    ).toEqual(11);
  });

  it('returns a new selection with whole group removed only if the whole group is present already', () => {
    const selection = {
      'RHEL-8': {
        'RHEL-8': false,
        'RHEL-8-8.0': true,
        'RHEL-8-8.1': true,
        'RHEL-8-8.2': true,
        'RHEL-8-8.3': true,
        'RHEL-8-8.4': true,
        'RHEL-8-8.5': true,
        'RHEL-8-8.6': true,
        'RHEL-8-8.7': true,
        'RHEL-8-8.8': true,
        'RHEL-8-8.9': true,
      },
    };

    expect(appendGroupSelection(selection, operatingSystemsGroups)).toEqual({});
  });

  it('it allows deselecting a minor version even when major is set to true (which actually should be null)', () => {
    const selection = {
      'RHEL-8': {
        'RHEL-8': true,
        'RHEL-8-8.0': true,
        'RHEL-8-8.1': true,
        'RHEL-8-8.2': true,
        'RHEL-8-8.3': true,
        'RHEL-8-8.4': true,
        'RHEL-8-8.5': true,
        'RHEL-8-8.6': true,
        'RHEL-8-8.7': true,
        'RHEL-8-8.8': false,
        'RHEL-8-8.9': true,
      },
    };

    expect(appendGroupSelection(selection, operatingSystemsGroups)).toEqual({
      'RHEL-8': {
        'RHEL-8': null,
        'RHEL-8-8.0': true,
        'RHEL-8-8.1': true,
        'RHEL-8-8.2': true,
        'RHEL-8-8.3': true,
        'RHEL-8-8.4': true,
        'RHEL-8-8.5': true,
        'RHEL-8-8.6': true,
        'RHEL-8-8.7': true,
        'RHEL-8-8.9': true,
      },
    });
  });

  it('it allows deselecting a minor version even when major is set to true (which actually should be null)', () => {
    const selection = {
      'RHEL-8': {
        'RHEL-8-8.0': true,
        'RHEL-8-8.1': true,
        'RHEL-8-8.2': true,
        'RHEL-8-8.3': true,
        'RHEL-8-8.4': true,
        'RHEL-8-8.5': true,
        'RHEL-8-8.6': true,
        'RHEL-8-8.7': true,
        'RHEL-8-8.8': true,
        'RHEL-8-8.9': true,
      },
    };

    expect(appendGroupSelection(selection, operatingSystemsGroups)).toEqual({
      'RHEL-8': {
        'RHEL-8': true,
        'RHEL-8-8.0': true,
        'RHEL-8-8.1': true,
        'RHEL-8-8.2': true,
        'RHEL-8-8.3': true,
        'RHEL-8-8.4': true,
        'RHEL-8-8.5': true,
        'RHEL-8-8.6': true,
        'RHEL-8-8.7': true,
        'RHEL-8-8.8': true,
        'RHEL-8-8.9': true,
      },
    });
  });
});
