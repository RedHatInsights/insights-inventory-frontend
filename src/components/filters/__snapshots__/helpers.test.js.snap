// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toOsFilterGroups returns ... when OSes are loaded 1`] = `
[
  {
    "groupSelectable": true,
    "items": [
      {
        "groupKey": "RHEL-9",
        "label": "RHEL 9.19",
        "type": "checkbox",
        "value": "RHEL-9-9.19",
      },
      {
        "groupKey": "RHEL-9",
        "label": "RHEL 9.18",
        "type": "checkbox",
        "value": "RHEL-9-9.18",
      },
      {
        "groupKey": "RHEL-9",
        "label": "RHEL 9.17",
        "type": "checkbox",
        "value": "RHEL-9-9.17",
      },
      {
        "groupKey": "RHEL-9",
        "label": "RHEL 9.16",
        "type": "checkbox",
        "value": "RHEL-9-9.16",
      },
      {
        "groupKey": "RHEL-9",
        "label": "RHEL 9.15",
        "type": "checkbox",
        "value": "RHEL-9-9.15",
      },
      {
        "groupKey": "RHEL-9",
        "label": "RHEL 9.14",
        "type": "checkbox",
        "value": "RHEL-9-9.14",
      },
      {
        "groupKey": "RHEL-9",
        "label": "RHEL 9.13",
        "type": "checkbox",
        "value": "RHEL-9-9.13",
      },
      {
        "groupKey": "RHEL-9",
        "label": "RHEL 9.12",
        "type": "checkbox",
        "value": "RHEL-9-9.12",
      },
      {
        "groupKey": "RHEL-9",
        "label": "RHEL 9.11",
        "type": "checkbox",
        "value": "RHEL-9-9.11",
      },
      {
        "groupKey": "RHEL-9",
        "label": "RHEL 9.10",
        "type": "checkbox",
        "value": "RHEL-9-9.10",
      },
      {
        "groupKey": "RHEL-9",
        "label": "RHEL 9.9",
        "type": "checkbox",
        "value": "RHEL-9-9.9",
      },
      {
        "groupKey": "RHEL-9",
        "label": "RHEL 9.8",
        "type": "checkbox",
        "value": "RHEL-9-9.8",
      },
      {
        "groupKey": "RHEL-9",
        "label": "RHEL 9.7",
        "type": "checkbox",
        "value": "RHEL-9-9.7",
      },
      {
        "groupKey": "RHEL-9",
        "label": "RHEL 9.6",
        "type": "checkbox",
        "value": "RHEL-9-9.6",
      },
      {
        "groupKey": "RHEL-9",
        "label": "RHEL 9.5",
        "type": "checkbox",
        "value": "RHEL-9-9.5",
      },
      {
        "groupKey": "RHEL-9",
        "label": "RHEL 9.4",
        "type": "checkbox",
        "value": "RHEL-9-9.4",
      },
      {
        "groupKey": "RHEL-9",
        "label": "RHEL 9.3",
        "type": "checkbox",
        "value": "RHEL-9-9.3",
      },
      {
        "groupKey": "RHEL-9",
        "label": "RHEL 9.2",
        "type": "checkbox",
        "value": "RHEL-9-9.2",
      },
      {
        "groupKey": "RHEL-9",
        "label": "RHEL 9.1",
        "type": "checkbox",
        "value": "RHEL-9-9.1",
      },
      {
        "groupKey": "RHEL-9",
        "label": "RHEL 9.0",
        "type": "checkbox",
        "value": "RHEL-9-9.0",
      },
    ],
    "label": "RHEL 9",
    "major": 9,
    "type": "checkbox",
    "value": "RHEL-9",
  },
  {
    "groupSelectable": true,
    "items": [
      {
        "groupKey": "CentOS-Linux-8",
        "label": "CentOS Linux 8.19",
        "type": "checkbox",
        "value": "CentOS-Linux-8-8.19",
      },
      {
        "groupKey": "CentOS-Linux-8",
        "label": "CentOS Linux 8.18",
        "type": "checkbox",
        "value": "CentOS-Linux-8-8.18",
      },
      {
        "groupKey": "CentOS-Linux-8",
        "label": "CentOS Linux 8.17",
        "type": "checkbox",
        "value": "CentOS-Linux-8-8.17",
      },
      {
        "groupKey": "CentOS-Linux-8",
        "label": "CentOS Linux 8.16",
        "type": "checkbox",
        "value": "CentOS-Linux-8-8.16",
      },
      {
        "groupKey": "CentOS-Linux-8",
        "label": "CentOS Linux 8.15",
        "type": "checkbox",
        "value": "CentOS-Linux-8-8.15",
      },
      {
        "groupKey": "CentOS-Linux-8",
        "label": "CentOS Linux 8.14",
        "type": "checkbox",
        "value": "CentOS-Linux-8-8.14",
      },
      {
        "groupKey": "CentOS-Linux-8",
        "label": "CentOS Linux 8.13",
        "type": "checkbox",
        "value": "CentOS-Linux-8-8.13",
      },
      {
        "groupKey": "CentOS-Linux-8",
        "label": "CentOS Linux 8.12",
        "type": "checkbox",
        "value": "CentOS-Linux-8-8.12",
      },
      {
        "groupKey": "CentOS-Linux-8",
        "label": "CentOS Linux 8.11",
        "type": "checkbox",
        "value": "CentOS-Linux-8-8.11",
      },
      {
        "groupKey": "CentOS-Linux-8",
        "label": "CentOS Linux 8.10",
        "type": "checkbox",
        "value": "CentOS-Linux-8-8.10",
      },
      {
        "groupKey": "CentOS-Linux-8",
        "label": "CentOS Linux 8.9",
        "type": "checkbox",
        "value": "CentOS-Linux-8-8.9",
      },
      {
        "groupKey": "CentOS-Linux-8",
        "label": "CentOS Linux 8.8",
        "type": "checkbox",
        "value": "CentOS-Linux-8-8.8",
      },
      {
        "groupKey": "CentOS-Linux-8",
        "label": "CentOS Linux 8.7",
        "type": "checkbox",
        "value": "CentOS-Linux-8-8.7",
      },
      {
        "groupKey": "CentOS-Linux-8",
        "label": "CentOS Linux 8.6",
        "type": "checkbox",
        "value": "CentOS-Linux-8-8.6",
      },
      {
        "groupKey": "CentOS-Linux-8",
        "label": "CentOS Linux 8.5",
        "type": "checkbox",
        "value": "CentOS-Linux-8-8.5",
      },
      {
        "groupKey": "CentOS-Linux-8",
        "label": "CentOS Linux 8.4",
        "type": "checkbox",
        "value": "CentOS-Linux-8-8.4",
      },
      {
        "groupKey": "CentOS-Linux-8",
        "label": "CentOS Linux 8.3",
        "type": "checkbox",
        "value": "CentOS-Linux-8-8.3",
      },
      {
        "groupKey": "CentOS-Linux-8",
        "label": "CentOS Linux 8.2",
        "type": "checkbox",
        "value": "CentOS-Linux-8-8.2",
      },
      {
        "groupKey": "CentOS-Linux-8",
        "label": "CentOS Linux 8.1",
        "type": "checkbox",
        "value": "CentOS-Linux-8-8.1",
      },
      {
        "groupKey": "CentOS-Linux-8",
        "label": "CentOS Linux 8.0",
        "type": "checkbox",
        "value": "CentOS-Linux-8-8.0",
      },
    ],
    "label": "CentOS Linux 8",
    "major": 8,
    "type": "checkbox",
    "value": "CentOS-Linux-8",
  },
  {
    "groupSelectable": true,
    "items": [
      {
        "groupKey": "RHEL-8",
        "label": "RHEL 8.19",
        "type": "checkbox",
        "value": "RHEL-8-8.19",
      },
      {
        "groupKey": "RHEL-8",
        "label": "RHEL 8.18",
        "type": "checkbox",
        "value": "RHEL-8-8.18",
      },
      {
        "groupKey": "RHEL-8",
        "label": "RHEL 8.17",
        "type": "checkbox",
        "value": "RHEL-8-8.17",
      },
      {
        "groupKey": "RHEL-8",
        "label": "RHEL 8.16",
        "type": "checkbox",
        "value": "RHEL-8-8.16",
      },
      {
        "groupKey": "RHEL-8",
        "label": "RHEL 8.15",
        "type": "checkbox",
        "value": "RHEL-8-8.15",
      },
      {
        "groupKey": "RHEL-8",
        "label": "RHEL 8.14",
        "type": "checkbox",
        "value": "RHEL-8-8.14",
      },
      {
        "groupKey": "RHEL-8",
        "label": "RHEL 8.13",
        "type": "checkbox",
        "value": "RHEL-8-8.13",
      },
      {
        "groupKey": "RHEL-8",
        "label": "RHEL 8.12",
        "type": "checkbox",
        "value": "RHEL-8-8.12",
      },
      {
        "groupKey": "RHEL-8",
        "label": "RHEL 8.11",
        "type": "checkbox",
        "value": "RHEL-8-8.11",
      },
      {
        "groupKey": "RHEL-8",
        "label": "RHEL 8.10",
        "type": "checkbox",
        "value": "RHEL-8-8.10",
      },
      {
        "groupKey": "RHEL-8",
        "label": "RHEL 8.9",
        "type": "checkbox",
        "value": "RHEL-8-8.9",
      },
      {
        "groupKey": "RHEL-8",
        "label": "RHEL 8.8",
        "type": "checkbox",
        "value": "RHEL-8-8.8",
      },
      {
        "groupKey": "RHEL-8",
        "label": "RHEL 8.7",
        "type": "checkbox",
        "value": "RHEL-8-8.7",
      },
      {
        "groupKey": "RHEL-8",
        "label": "RHEL 8.6",
        "type": "checkbox",
        "value": "RHEL-8-8.6",
      },
      {
        "groupKey": "RHEL-8",
        "label": "RHEL 8.5",
        "type": "checkbox",
        "value": "RHEL-8-8.5",
      },
      {
        "groupKey": "RHEL-8",
        "label": "RHEL 8.4",
        "type": "checkbox",
        "value": "RHEL-8-8.4",
      },
      {
        "groupKey": "RHEL-8",
        "label": "RHEL 8.3",
        "type": "checkbox",
        "value": "RHEL-8-8.3",
      },
      {
        "groupKey": "RHEL-8",
        "label": "RHEL 8.2",
        "type": "checkbox",
        "value": "RHEL-8-8.2",
      },
      {
        "groupKey": "RHEL-8",
        "label": "RHEL 8.1",
        "type": "checkbox",
        "value": "RHEL-8-8.1",
      },
      {
        "groupKey": "RHEL-8",
        "label": "RHEL 8.0",
        "type": "checkbox",
        "value": "RHEL-8-8.0",
      },
    ],
    "label": "RHEL 8",
    "major": 8,
    "type": "checkbox",
    "value": "RHEL-8",
  },
  {
    "groupSelectable": true,
    "items": [
      {
        "groupKey": "CentOS-Linux-7",
        "label": "CentOS Linux 7.19",
        "type": "checkbox",
        "value": "CentOS-Linux-7-7.19",
      },
      {
        "groupKey": "CentOS-Linux-7",
        "label": "CentOS Linux 7.18",
        "type": "checkbox",
        "value": "CentOS-Linux-7-7.18",
      },
      {
        "groupKey": "CentOS-Linux-7",
        "label": "CentOS Linux 7.17",
        "type": "checkbox",
        "value": "CentOS-Linux-7-7.17",
      },
      {
        "groupKey": "CentOS-Linux-7",
        "label": "CentOS Linux 7.16",
        "type": "checkbox",
        "value": "CentOS-Linux-7-7.16",
      },
      {
        "groupKey": "CentOS-Linux-7",
        "label": "CentOS Linux 7.15",
        "type": "checkbox",
        "value": "CentOS-Linux-7-7.15",
      },
      {
        "groupKey": "CentOS-Linux-7",
        "label": "CentOS Linux 7.14",
        "type": "checkbox",
        "value": "CentOS-Linux-7-7.14",
      },
      {
        "groupKey": "CentOS-Linux-7",
        "label": "CentOS Linux 7.13",
        "type": "checkbox",
        "value": "CentOS-Linux-7-7.13",
      },
      {
        "groupKey": "CentOS-Linux-7",
        "label": "CentOS Linux 7.12",
        "type": "checkbox",
        "value": "CentOS-Linux-7-7.12",
      },
      {
        "groupKey": "CentOS-Linux-7",
        "label": "CentOS Linux 7.11",
        "type": "checkbox",
        "value": "CentOS-Linux-7-7.11",
      },
      {
        "groupKey": "CentOS-Linux-7",
        "label": "CentOS Linux 7.10",
        "type": "checkbox",
        "value": "CentOS-Linux-7-7.10",
      },
      {
        "groupKey": "CentOS-Linux-7",
        "label": "CentOS Linux 7.9",
        "type": "checkbox",
        "value": "CentOS-Linux-7-7.9",
      },
      {
        "groupKey": "CentOS-Linux-7",
        "label": "CentOS Linux 7.8",
        "type": "checkbox",
        "value": "CentOS-Linux-7-7.8",
      },
      {
        "groupKey": "CentOS-Linux-7",
        "label": "CentOS Linux 7.7",
        "type": "checkbox",
        "value": "CentOS-Linux-7-7.7",
      },
      {
        "groupKey": "CentOS-Linux-7",
        "label": "CentOS Linux 7.6",
        "type": "checkbox",
        "value": "CentOS-Linux-7-7.6",
      },
      {
        "groupKey": "CentOS-Linux-7",
        "label": "CentOS Linux 7.5",
        "type": "checkbox",
        "value": "CentOS-Linux-7-7.5",
      },
      {
        "groupKey": "CentOS-Linux-7",
        "label": "CentOS Linux 7.4",
        "type": "checkbox",
        "value": "CentOS-Linux-7-7.4",
      },
      {
        "groupKey": "CentOS-Linux-7",
        "label": "CentOS Linux 7.3",
        "type": "checkbox",
        "value": "CentOS-Linux-7-7.3",
      },
      {
        "groupKey": "CentOS-Linux-7",
        "label": "CentOS Linux 7.2",
        "type": "checkbox",
        "value": "CentOS-Linux-7-7.2",
      },
      {
        "groupKey": "CentOS-Linux-7",
        "label": "CentOS Linux 7.1",
        "type": "checkbox",
        "value": "CentOS-Linux-7-7.1",
      },
      {
        "groupKey": "CentOS-Linux-7",
        "label": "CentOS Linux 7.0",
        "type": "checkbox",
        "value": "CentOS-Linux-7-7.0",
      },
    ],
    "label": "CentOS Linux 7",
    "major": 7,
    "type": "checkbox",
    "value": "CentOS-Linux-7",
  },
  {
    "groupSelectable": true,
    "items": [
      {
        "groupKey": "RHEL-7",
        "label": "RHEL 7.19",
        "type": "checkbox",
        "value": "RHEL-7-7.19",
      },
      {
        "groupKey": "RHEL-7",
        "label": "RHEL 7.18",
        "type": "checkbox",
        "value": "RHEL-7-7.18",
      },
      {
        "groupKey": "RHEL-7",
        "label": "RHEL 7.17",
        "type": "checkbox",
        "value": "RHEL-7-7.17",
      },
      {
        "groupKey": "RHEL-7",
        "label": "RHEL 7.16",
        "type": "checkbox",
        "value": "RHEL-7-7.16",
      },
      {
        "groupKey": "RHEL-7",
        "label": "RHEL 7.15",
        "type": "checkbox",
        "value": "RHEL-7-7.15",
      },
      {
        "groupKey": "RHEL-7",
        "label": "RHEL 7.14",
        "type": "checkbox",
        "value": "RHEL-7-7.14",
      },
      {
        "groupKey": "RHEL-7",
        "label": "RHEL 7.13",
        "type": "checkbox",
        "value": "RHEL-7-7.13",
      },
      {
        "groupKey": "RHEL-7",
        "label": "RHEL 7.12",
        "type": "checkbox",
        "value": "RHEL-7-7.12",
      },
      {
        "groupKey": "RHEL-7",
        "label": "RHEL 7.11",
        "type": "checkbox",
        "value": "RHEL-7-7.11",
      },
      {
        "groupKey": "RHEL-7",
        "label": "RHEL 7.10",
        "type": "checkbox",
        "value": "RHEL-7-7.10",
      },
      {
        "groupKey": "RHEL-7",
        "label": "RHEL 7.9",
        "type": "checkbox",
        "value": "RHEL-7-7.9",
      },
      {
        "groupKey": "RHEL-7",
        "label": "RHEL 7.8",
        "type": "checkbox",
        "value": "RHEL-7-7.8",
      },
      {
        "groupKey": "RHEL-7",
        "label": "RHEL 7.7",
        "type": "checkbox",
        "value": "RHEL-7-7.7",
      },
      {
        "groupKey": "RHEL-7",
        "label": "RHEL 7.6",
        "type": "checkbox",
        "value": "RHEL-7-7.6",
      },
      {
        "groupKey": "RHEL-7",
        "label": "RHEL 7.5",
        "type": "checkbox",
        "value": "RHEL-7-7.5",
      },
      {
        "groupKey": "RHEL-7",
        "label": "RHEL 7.4",
        "type": "checkbox",
        "value": "RHEL-7-7.4",
      },
      {
        "groupKey": "RHEL-7",
        "label": "RHEL 7.3",
        "type": "checkbox",
        "value": "RHEL-7-7.3",
      },
      {
        "groupKey": "RHEL-7",
        "label": "RHEL 7.2",
        "type": "checkbox",
        "value": "RHEL-7-7.2",
      },
      {
        "groupKey": "RHEL-7",
        "label": "RHEL 7.1",
        "type": "checkbox",
        "value": "RHEL-7-7.1",
      },
      {
        "groupKey": "RHEL-7",
        "label": "RHEL 7.0",
        "type": "checkbox",
        "value": "RHEL-7-7.0",
      },
    ],
    "label": "RHEL 7",
    "major": 7,
    "type": "checkbox",
    "value": "RHEL-7",
  },
]
`;
